index.js

```diff
@@ -133,7 +128,7 @@ function getOptions(uri, o, method) {
     }
 
     o.agentClass = /^https/.test(o.uri) ? httpsAgentClass : httpAgentClass;
-    o.agentOptions = agentOptions;
+    o.agentOptions = Object.assign(defaultAgentOptions, o.agentOptions || {});
```

Comment:
This changes `defaultAgentOptions`. Did you mean(TM): `Object.assign({}, defaultAgentOptions, o.agentOptions)` ?